Sources/ServiceLifecycle/ServiceGroup.swift

```diff
@@ -33,6 +33,10 @@ public actor ServiceGroup: Sendable {
     private let logger: Logger
     /// The logging configuration.
     private let loggingConfiguration: ServiceGroupConfiguration.LoggingConfiguration
+    /// The maximum amount of time that graceful shutdown is allowed to take.
+    private var maximumGracefulShutdownDuration: (secondsComponent: Int64, attosecondsComponent: Int64)?
+    /// The maximum amount of time that task cancellation is allowed to take.
+    private var maximumCancellationDuration: (secondsComponent: Int64, attosecondsComponent: Int64)?
```

Comment:
Why are these mutable? Also complete nit, but you'll save some repetition converting between the tuples and `Duration` if you define your own `_Duration`.

Reply:
Good catch with the `var`. Was a bad copy and paste. I decided not to introduce a custom type for now. if we do more `Duration` based APIs we should come back to that.